app/src/store/voice/accessToken.ts

```diff
@@ -0,0 +1,125 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { fetch, defaultUrl } from '../../util/fetch';
+import { settlePromise } from '../../util/settlePromise';
+import { type AsyncStoreSlice, type State, type Dispatch } from '../app';
+
+/**
+ * Actions
+ */
+export type GetAccessTokenRejectValue =
+  | {
+      reason: 'USER_NOT_FULFILLED';
+    }
+  | {
+      reason: 'FETCH_ERROR';
+      error: any;
+    }
+  | {
+      reason: 'TOKEN_RESPONSE_NOT_OK';
+      statusCode: number;
+    }
+  | {
+      reason: 'FETCH_TEXT_ERROR';
+      error: any;
+    };
+
+export const getAccessToken = createAsyncThunk<
+  string,
+  void,
+  {
+    state: State;
+    dispatch: Dispatch;
+    rejectValue: GetAccessTokenRejectValue;
+  }
+>('voice/getAccessToken', async (_, { getState, rejectWithValue }) => {
+  const user = getState().voice.user;
+  if (user?.status !== 'fulfilled') {
+    return rejectWithValue({ reason: 'USER_NOT_FULFILLED' });
+  }
+
+  const fetchResult = await settlePromise(
+    fetch(`${defaultUrl}/token`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${user.accessToken}`,
+      },
+    }),
+  );
+  if (fetchResult.status === 'rejected') {
+    return rejectWithValue({
+      reason: 'FETCH_ERROR',
+      error: fetchResult.reason,
+    });
+  }
+
+  const tokenResponse = fetchResult.value;
+  if (!tokenResponse.ok) {
+    return rejectWithValue({
+      reason: 'TOKEN_RESPONSE_NOT_OK',
+      statusCode: tokenResponse.status,
+    });
+  }
+
+  const tokenTextResult = await settlePromise(tokenResponse.text());
+  if (tokenTextResult.status === 'rejected') {
+    return rejectWithValue({
+      reason: 'FETCH_TEXT_ERROR',
+      error: tokenTextResult.reason,
+    });
+  }
+
+  const token = tokenTextResult.value;
+  return token;
+});
+
+/**
+ * Slice
```

Comment:
I don't think this "slice" comment is necessary since line 79 says `AsyncStoreSlice`